docs/StyleGuide.md

```diff
@@ -0,0 +1,1142 @@
+  and distracts. Consistency is also a helpful guiding factor when deciding
+  on style guidelines in the first place; sometimes a choice between formatting
+  something this way or that way seems arbitrary in isolation, but becomes
+  clearer when seen as part of a coherent whole.
+
+## Formatting
+
+We now list the formatting rules we have converged on. As these have grown
+organically, not all code follows these rules. When touching some existing code,
+we in general recommend sticking to the existing style, but when it differs from
+the rules below, it is good practice to update the code's style to match them.
+
+2. __Indentation__: we indent by 2 spaces.
+
+   *Why:* to avoid wasting horizontal screen space.
+
+   Some consequences of this rule:
+
+   a. The `where` clause of a function body is indented 2 spaces from the left
+      margin, and the function body is indented 2 spaces from the `where`:
+
+      ```haskell
+      foo x y z =
+          ..
+        where
+          a = ..
+      ```
+
+      The `where` keyword acts as a separator between the body and the bindings.
+      Keeping them at the same indentation level would make it hard to see where
+      the body ends.
+
+      We stick with this indentation even if the `where` clause is not present,
+      just to avoid unnecessary changes when a `where` clause is added.
+
+   b. We indent record `data` and `newtype` definitions as follows:
+
+      ```haskell
+      data Foo = Foo
+          { fooBar      :: Int
+          , fooArgument :: Bool
+          }
+        deriving (Show, Eq)
+
+      newtype Foo = Foo { unFoo :: Int }
+        deriving (Show, Eq)
+      ```
+
+      The `deriving` is indented from the left margin, and the constructors
+      are indented from the `deriving` clause. This provides a consistent
+      style for data types with multiple constructors (see below).
+
+      Multiple deriving clauses using `DerivingStrategies` are aligned:
+
+      ```haskell
+      data Foo = Foo
+          { fooBar      :: Int
+          , fooArgument :: Bool
+          }
+        deriving stock    (Show, Eq, Generic)
+        deriving anyclass (NoThunks, NFData)
+
+      newtype Foo = Foo {
+            unFoo :: Int
+          }
+        deriving stock   (Show)
+        deriving newtype (Eq)
+        deriving NoThunks via InspectHeapNamed "Foo" Foo
+      ```
+
+      Parentheses around a singleton list of classes are optional.
+
+      Records that fit onto a single line can be formatted like this:
+
+      ```haskell
+      data Foo = X {foo :: A, bar :: B} | Y
+      ```
+
+      We aren't very consistent with our data declarations, following the local
+      style is always a good choice.
+
+   c. We indent `data` definitions with multiple constructors as follows:
+
+      ```haskell
+      data Foo =
+          Bar Int Int
+        | Baz
+            Int
+            Int
+            (Maybe Bool)
+            [Foo]
+      ```
```

Comment:
I would prefer

```haskell
data Foo = Bar Int Int
         | Baz Int
               Int
               (Maybe Bool)
               [Foo]
```

If space allows, I just find it more readable with the bar aligned with the equals sign and the first argument in the same line as the constructor but not always that is possible due to space constraints

Reply:
I think this one with haddocks:

```haskell
data Foo
    = -- | Bar
      --
      Bar Int Int
      
    | -- | Baz
      --
      Baz Int
          Int
          (Maybe Bool)
          [Foo]
```

will be more consistent than one where `=` on the `data Foo =` line.

Reply:
Yes I agree!